main/src/cgeo/geocaching/sorting/TargetDistanceComparator.java

```diff
@@ -0,0 +1,18 @@
+package cgeo.geocaching.sorting;
+
+import cgeo.geocaching.location.Geopoint;
+
+/**
+ * sorts caches by distance to given target position
+ *
+ */
+public class TargetDistanceComparator extends AbstractDistanceComparator {
+
+    public static final TargetDistanceComparator DISTANCE_TO_TARGET = new TargetDistanceComparator();
```

Comment:
A construct using a static member is not necessary for target distance connector and should be removed. 

I assume this was copied from the approach for distance to current position. This is used to update the distance sort when the user moves and thus distance to caches may change.

For target distance however, the target coord and distance will remain constant, thus the target const can simply be passed in constructor. Usage of an ugly global var is not necessary.

Reply:
We however need an initialized instance here... I can rename both global vars to "INSTANCE", but this will not solve the general problem. Any idea?



![grafik](https://user-images.githubusercontent.com/64581222/133933987-32745433-82bf-4e1f-94c2-2cd4a23ade7b.png)

Reply:
I've just renamed both to "INSTANCE" for consistency. That's exactly how it is done for all other comparators as well. But I'm open for suggestions how to archive it differently...

Reply:
@fm-sys see my comment on another method for solution suggestion . Alternatively, create a new instance of the comparator here. For this to work, the target coord needs to get a member of the class (but I would personally prefer my other suggestion)

In all the cases where INSTANCE is used, the comparator is not parameterized, so usage of a constant instance is fine. TargetDistanceComparator however needs to be parameterized.

 "Distance to user" is special due to the ever-changing user-coord and thus necessity to resort (but on the other hand there can always be one user-coordinate globally at any point in time).